torch/autograd/gradcheck.py

```diff
@@ -207,7 +207,8 @@ def gradcheck(
     rtol: float = 1e-3,
     raise_exception: bool = True,
     check_sparse_nnz: bool = False,
-    nondet_tol: float = 0.0
+    nondet_tol: float = 0.0,
+    check_undefined_grad: bool = True
```

Comment:
Setting this default to True is a BC-breaking change I'm afraid.
I think it is fine but we want to make sure that it is simple for the user to identify why they will see different behavior.

In particular, could you add a warning below just before raising any error stating that this is the new undefined_grad check in gradcheck and that they can turn it off with this argument if they don't want it.

Reply:
Done